### PR TITLE
Keep Scoop package listings working when scoop commands fail

### LINKED ISSUE #5262: [IMPROVEMENT] Don't fail to scoop list operations if scoop update fails
### Please confirm these before moving forward.

- [x] I have searched for my feature proposal and have not found a work-in-progress/duplicate/resolved/discarded issue.
- [x] This improvement refers to an existing feature. If you want to suggest a new feature, please use [this template](https://github.com/Devolutions/UniGetUI/issues/new?labels=new-feature&projects=&template=feature-request.yml&title=%5BFEATURE+REQUEST%5D+%28Enter+your+description+here%29).
- [x] This improvement is not a bug. If you want to report a bug, please use [this template](https://github.com/Devolutions/UniGetUI/issues/new?labels=bug&projects=&template=bug-issue.yml&title=%5BBUG%5D+%28Enter+your+description+here%29).

### Describe the improvement

I've noticed the UI failed to list the installed packages from Scoop even though I could install with Scoop. I then noticed that `scoop update` failed since one of the bucket was broken
I'd like the failure of that operation to have no side effects, i.e., the list should still be populated
Understandably, the Software updates can't be populated for Scoop, but this shouldn't block just the app list?

(would be nice if GUI also showed some warning, though the logs had the errors)

### Describe how this improvement could help users

Will delay the need to fix Scoop

### PR DESCRIPTION
**Pull request by GabrielDuf**
This pull request refactors how process output is read and logged in the Scoop package manager integration, improving reliability when dealing with processes that produce large amounts of output, especially on standard error. It introduces a new `ScoopProcess` helper class to centralize and streamline process output handling, replaces duplicated code with calls to this helper, and adds robust timeout and error handling. Comprehensive tests are also added to ensure the new logic handles edge cases like pipe buffer overflows.

**Process Output Handling Improvements**
* Introduced the `ScoopProcess` helper class to encapsulate logic for reading process standard output and error streams, supporting both line-by-line and full-output reads, and ensuring proper logging and process cleanup. This prevents deadlocks and handles large error outputs robustly.
* Refactored all usages in `Scoop.cs`, `ScoopPkgDetailsHelper.cs`, and `ScoopSourceHelper.cs` to use `ScoopProcess.ReadLines` and `ScoopProcess.ReadToEnd` instead of duplicating output reading and logging logic, simplifying these methods and improving reliability. 

**Timeouts and Error Handling**
* Enhanced version probing and stream draining with explicit timeouts and error logging in `_loadManagerVersion`, preventing hangs and reporting issues if the process does not respond in time. 
* Improved the `RunListingTaskWithTimeout` logic in `PackageManager.cs` to allow disabling timeouts only when appropriate, and added a `RefreshPackageIndexesSafely` method to handle exceptions gracefully and warn the user if index refresh fails. 

**Testing Enhancements**
* Added a `TestProcessTaskLogger` fake for unit testing process output logging.
* Added comprehensive tests in `ScoopProcessTests` to verify that reading process output works even when the standard error stream is flooded, ensuring the new logic is robust against pipe buffer limitations.

### CODE REVIEW

File: src/UniGetUI.PackageEngine.PackageManagerClasses/Manager/PackageManager.cs [hunk 1]
```diff
@@ -312,7 +343,10 @@ private T RunListingTaskWithTimeout<T>(Func<T> method, string taskName)
 
             if (!task.Wait(TimeSpan.FromSeconds(PackageListingTaskTimeout)))
             {
-                if (!Settings.Get(Settings.K.DisableTimeoutOnPackageListingTasks))
+                if (
+                    !allowDisablingTimeout
+                    || !Settings.Get(Settings.K.DisableTimeoutOnPackageListingTasks)
```
[the remark this answers is not in the document]
Comment (GabrielDuf) [reply src/UniGetUI.PackageEngine.PackageManagerClasses/Manager/PackageManager.cs:348]: Fixed in [aad9ed6](https://github.com/Devolutions/UniGetUI/pull/5321/commits/aad9ed6256fff163bc7a700a2d07021a8ea88057)

File: src/UniGetUI.PackageEngine.Managers.Scoop/Scoop.cs [hunk 2]
```diff
@@ -391,10 +394,7 @@ protected override IReadOnlyList<Package> FindPackages_UnSafe(string query)
                     proc
                 );
                 proc.Start();
-                aux_logger.AddToStdOut(proc.StandardOutput.ReadToEnd());
-                aux_logger.AddToStdErr(proc.StandardError.ReadToEnd());
-                proc.WaitForExit();
-                aux_logger.Close(proc.ExitCode);
+                ScoopProcess.ReadToEnd(proc, aux_logger);
```
[the remark this answers is not in the document]
Comment (GabrielDuf) [reply src/UniGetUI.PackageEngine.Managers.Scoop/Scoop.cs:397]: Line 503 already has `RegisterListingProcess(p)` — that's `RefreshPackageIndexes`, so no change is needed there.

Line 472 (`scoop list`) is left unregistered on purpose. It's `TaskRecycler`-backed, and `GetAvailableUpdates_UnSafe` attaches to the same recycled task, so the installed loader and the updates loader routinely share one process. Registering it would let the updates listing's timeout kill a process the installed loader is waiting on, and the installed list would come back empty despite having had plenty of time — trading a leaked process for a wrong result, which is what this PR is trying to remove.

Line 397 is the `scoop-search` bootstrap install, not a query. `KillListingProcesses` kills the whole tree, so a legitimate install that just takes over 60s on a slow connection would be torn down mid-download and could leave a broken app dir. Letting it finish in the background means the next search picks the binary up. Registering it also wouldn't prevent the duplicate child described: the retry re-runs `FindPackages_UnSafe`, `Which("scoop-search.exe")` is still false, and a second install starts either way — killing the first just ensures neither completes. That's pre-existing, and the real fix is to not bootstrap a dependency install from inside a timed listing task; worth a separate issue.
